src/Validation/Validator.php

```diff
@@ -2532,11 +2539,11 @@ protected function _processRules(string $field, ValidationSet $rules, array $dat
         $errors = [];
         // Loading default provider in case there is none
         $this->getProvider('default');
-        $message = 'The provided value is invalid';
 
-        if ($this->_useI18n) {
-            $message = __d('cake', 'The provided value is invalid');
-        }
+        $message = $this->getValidationMessage(
+            'The provided value is invalid',
+            __d('cake', 'The provided value is invalid')
+        );
```

Comment:
This will work ok in the fullstack framework where `useI18n` is always true. However, it will create runtime errors for standalone usage of the validation package. We'd need to have the calls to `__d` guarded. The need of guards was why I suggested we use placeholders, but it doesn't sound like that will work as the verb conjugation will depend on the noun gender.

Another option could be to have a case ladder for the i18n messages :shrug: . It wouldn't be simple but validators could use `$message = $this->getValidationMessage('invalid')` and we could have a `match` like:

```php
if ($this->_useI18n) {
  return match ($type) {
    case 'invalid' => __d('cake', 'The provided value is invalid.'),
    case 'array' => __d('cake', 'The provided value is not an array.'),
  }
}

return match ($type)  {
 // non i18n messages here
}
```
It won't be pretty but it will avoid the runtime errors and allow full message translation. :shrug:

Reply:
I reverted the ``getValidationMessage()`` again as I think using the verbose style is still more readable and guards the calls to `__d`  appropriately.